### PR TITLE
Skip Pattern to exclude certains URLs

### PR DESCRIPTION
**Pull request by pavolloffay**

### CODE REVIEW

**Comment by pavolloffay** [comment]
build passed, and locally it says that `io.opentracing.contrib:opentracing-web-servlet-filter:jar:0.0.6` is not in central yet..

File: opentracing-spring-web-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfiguration.java [hunk 1]
```diff
@@ -26,12 +25,19 @@
...
+    public WebTracingConfiguration tracerAutoConfiguration() {
+        return WebTracingConfiguration.builder()
+                .withSkipPattern(WebTracingConfiguration.DEFAULT_SKIP_PATTERN).build();
+    }
+
+    @Bean
+    public FilterRegistrationBean tracingFilter(Tracer tracer, WebTracingConfiguration tracingConfiguration) {
         log.info("Creating " + FilterRegistrationBean.class.getSimpleName() + " bean with " +
-                TracingFilter.class + " mapped to " + "/*");
+                TracingFilter.class + " mapped to " + "/*, skip pattern is " + tracingConfiguration.skipPattern);
```
Comment (jpkrohling) [inline opentracing-spring-web-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfiguration.java]: You are subject to a NPE here. Perhaps you'd want to do a null-check?

File: opentracing-spring-web-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/autoconfig/WebTracingConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import java.util.regex.Pattern;
+
+/**
+ * @author Pavol Loffay
+ */
+public class WebTracingConfiguration {
+    public static final Pattern DEFAULT_SKIP_PATTERN = Pattern.compile("/health|/favicon\\.ico");
+
+    public final Pattern skipPattern;
```
Comment (jpkrohling) [inline opentracing-spring-web-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/autoconfig/WebTracingConfiguration.java]: Copy/paste what I said on IRC about this field being public:

> I have nothing against it per se, *but*, it's a common style to provide getters/setters to properties in Java. If you are deviating from what people expect, make sure to have a reason and document it

File: opentracing-spring-web-itest/common/src/test/java/io/opentracing/contrib/spring/web/interceptor/itest/common/AbstractBaseITests.java [hunk 3]
```diff
@@ -418,6 +417,16 @@ public void testControllerView() {
         assertLogEvents(span.logEntries(), Arrays.asList("preHandle", "afterCompletion"));
     }
 
+    @Test
+    public void testExcludePattern() throws InterruptedException {
+        {
+            getRestTemplate().getForEntity("/health", String.class);
+            Thread.sleep(505);
```
Comment (jpkrohling) [inline opentracing-spring-web-itest/common/src/test/java/io/opentracing/contrib/spring/web/interceptor/itest/common/AbstractBaseITests.java]: Why this wait?

Comment (jpkrohling) [inline opentracing-spring-web-itest/common/src/test/java/io/opentracing/contrib/spring/web/interceptor/itest/common/AbstractBaseITests.java]: And more intriguingly, why `505`?

File: opentracing-spring-web/src/test/java/io/opentracing/contrib/spring/web/client/TracingRestTemplateInterceptorTest.java [hunk 4]
```diff
@@ -99,7 +99,7 @@ public void testStandardTagsWithPort() {
         Assert.assertEquals("GET", mockSpan.tags().get(Tags.HTTP_METHOD.getKey()));
         Assert.assertEquals(url, mockSpan.tags().get(Tags.HTTP_URL.getKey()));
         Assert.assertEquals(200, mockSpan.tags().get(Tags.HTTP_STATUS.getKey()));
-        Assert.assertEquals((short)4000, mockSpan.tags().get(Tags.PEER_PORT.getKey()));
```
Comment (jpkrohling) [inline opentracing-spring-web/src/test/java/io/opentracing/contrib/spring/web/client/TracingRestTemplateInterceptorTest.java]: This is not related to this change, right ?

File: opentracing-spring-web-itest/common/src/test/java/io/opentracing/contrib/spring/web/interceptor/itest/common/AbstractBaseITests.java (hunk 3 above)
Comment (pavolloffay) [inline opentracing-spring-web-itest/common/src/test/java/io/opentracing/contrib/spring/web/interceptor/itest/common/AbstractBaseITests.java]: I know that you will ask :+1: . Just making sure that span is not created.

> And more intriguingly, why 505? 

It's a nice number but maybe 100ms would be enough.

File: opentracing-spring-web/src/test/java/io/opentracing/contrib/spring/web/client/TracingRestTemplateInterceptorTest.java (hunk 4 above)
Comment (pavolloffay) [inline opentracing-spring-web/src/test/java/io/opentracing/contrib/spring/web/client/TracingRestTemplateInterceptorTest.java]: It is, this bumps filter version which uses ot 0.21 where port has been changed to `int`.

Comment (jpkrohling) [inline opentracing-spring-web/src/test/java/io/opentracing/contrib/spring/web/client/TracingRestTemplateInterceptorTest.java]: I imagined so, just didn't notice the version bump.

File: opentracing-spring-web-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfiguration.java (hunk 1 above)
Comment (pavolloffay) [inline opentracing-spring-web-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/autoconfig/ServerTracingAutoConfiguration.java]: `tracingConfiguration` won't be `null` and string concat with null works:
http://stackoverflow.com/questions/4260723/concatenating-null-strings-in-java
https://docs.oracle.com/javase/specs/jls/se8/html/jls-5.html#jls-5.1.11

File: opentracing-spring-web-itest/common/src/test/java/io/opentracing/contrib/spring/web/interceptor/itest/common/AbstractBaseITests.java (hunk 3 above)
Comment (jpkrohling) [inline opentracing-spring-web-itest/common/src/test/java/io/opentracing/contrib/spring/web/interceptor/itest/common/AbstractBaseITests.java]: Isn't there a better way to do that? Like, using Powermock to ensure a method is *not* called?

File: opentracing-spring-web-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/autoconfig/WebTracingConfiguration.java (hunk 2 above)
Comment (pavolloffay) [inline opentracing-spring-web-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/autoconfig/WebTracingConfiguration.java]: I have changed it to getter. The reason for this is that java is too verbose for this simple props.

**Comment by pavolloffay** [comment]
@jpkrohling thanks!
